--- a/Makefile
+++ b/Makefile
@@ -28,6 +28,7 @@ RCSVD = \
 	sv.d/misc \
 	sv.d/mount-all \
 	sv.d/net-lo \
+	sv.d/network \
 	sv.d/random-seed \
 	sv.d/remount-root \
 	sv.d/swap \
--- a/script/rc.conf.in
+++ b/script/rc.conf.in
@@ -26,3 +26,16 @@
 # /sys/fs/cgroup in hybrid or legacy mode
 
 # HAVE_CONTROLLER1_GROUPS="true"
+
+# The following options control the static network configuration
+#   - NET_INTERFACE: name of device (required)
+#   - NET_ADDRESS: IP address (required)
+#   - NET_NETMASK: subnet mask (optional, defaults to 255.255.255.0)
+#   - NET_BROADCAST: broadcast address (optional)
+#   - NET_GATEWAY: default route (required)
+
+# NET_INTERFACE=""
+# NET_ADDRESS=""
+# NET_NETMASK=""
+# NET_BROADCAST=""
+# NET_GATEWAY=""
--- /dev/null
+++ b/sv.d/network.in
@@ -0,0 +1,40 @@
+#!/bin/bash
+
+# sourcing our current rc.conf requires this to be a bash script
+. @RCLIBDIR@/functions
+. @RCDIR@/rc.conf
+
+case "$1" in
+    start)
+        stat_busy "Starting network"
+        if [[ -n $NET_INTERFACE && -n $NET_ADDRESS && -n $NET_GATEWAY ]]; then
+            ip link set dev "$NET_INTERFACE" up || stat_die network
+            ip addr add "$NET_ADDRESS/${NET_NETMASK:-24}" brd "${NET_BROADCAST:-+}" dev "$NET_INTERFACE" || stat_die network
+            ip route add default via "$NET_GATEWAY" || stat_die network
+            add_daemon network
+            stat_done network
+        else
+            stat_die network
+        fi
+        ;;
+    stop)
+        stat_busy "Stopping network"
+        if [[ -n $NET_INTERFACE ]]; then
+            ip addr flush dev "$NET_INTERFACE" || stat_die network
+            ip link set dev "$NET_INTERFACE" down || stat_die network
+            rm_daemon network
+            stat_done network
+        else
+            stat_die network
+        fi
+        ;;
+    restart)
+        $0 stop
+        sleep 1
+        $0 start
+        ;;
+    *)
+        echo "usage: $0 {start|stop|restart}"
+        exit 1
+        ;;
+esac
