--- a/script/3.in
+++ b/script/3.in
@@ -4,6 +4,8 @@ PATH=/usr/bin:/usr/sbin
 
 . @RCLIBDIR@/functions
 
+[ -x @RCDIR@/rc.shutdown ] && @RCDIR@/rc.shutdown
+
 stat_busy "Stop services ..."
 sv force-stop @RUNDIR@/service/*
 stat_done
@@ -11,8 +13,6 @@ stat_busy "Exit services ..."
 sv exit @RUNDIR@/service/*
 stat_done
 
-[ -x @RCDIR@/rc.shutdown ] && @RCDIR@/rc.shutdown
-
 if [ -e @RUNDIR@/reboot ]; then
     chmod 100 @RUNDIR@/reboot
 fi
--- a/script/rc.shutdown.in
+++ b/script/rc.shutdown.in
@@ -1,4 +1,4 @@
 # @RCDIR@/rc.shutdown -- rc.shutdown for Artix Linux
 #
 # Enter your custom commands here. It will be executed on stage 3
-# after stopping services.
+# before shutting down services.
