--- a/install/consolefont
+++ b/install/consolefont
@@ -3,10 +3,10 @@
 build() {
     # subshell to avoid namespace pollution
     (
-        [[ -s /etc/vconsole.conf ]] && . /etc/vconsole.conf
+        [[ -s /etc/conf.d/consolefont ]] && . /etc/conf.d/consolefont
 
         if [[ $FONT ]]; then
-            for file in "/usr/share/kbd/consolefonts/$FONT".@(fnt|psf?(u))?(.gz); do
+            for file in "/usr/share/consolefonts/$FONT".@(fnt|psf?(u))?(.gz); do
                 if [[ -e $file ]]; then
                     [[ $file =~ (\.(fnt|psfu?))(\.gz)?$ ]] && ext=${BASH_REMATCH[2]}
                     if [[ $file = *.gz ]]; then
