diff -Nuar mididings-20120419_/setup.py mididings-20120419/setup.py
--- mididings-20120419_/setup.py	2019-07-23 22:34:48.259922727 +0000
+++ mididings-20120419/setup.py	2019-07-23 22:41:37.433013086 +0000
@@ -46,14 +46,12 @@
             library_dirs.append(val)
 
 def boost_lib_name(lib):
-    vmaj = sys.version_info[0]
-    vmin = sys.version_info[1]
-
+    version = '%d%d' % sys.version_info[:2]
     for libdir in ('/usr/lib', '/usr/local/lib', '/usr/lib64', '/usr/local/lib64'):
-        for suffix in ('', '-mt'):
-            libname = 'lib%s-%d.%d%s.so' % (lib, vmaj, vmin, suffix)
+        for suffix in ('', version, '-mt', version + '-mt'):
+            libname = 'lib%s%s.so' % (lib, suffix)
             if os.path.isfile(os.path.join(libdir, libname)):
-                return '%s-%d.%d%s' % (lib, vmaj, vmin, suffix)
+                return lib + suffix
     return lib + '-mt'
 
 
