From 73a4d69ff8d3430c9f6c58fb1e1e2e7e255e72fb Mon Sep 17 00:00:00 2001
From: neeshy <neeshy@tfwno.gf>
Date: Wed, 24 May 2023 20:32:24 -0400
Subject: [PATCH 3/4] su: don't extraneously set shell when pflag is active

---
 su.c | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/su.c b/su.c
index 812aa43..b637a91 100644
--- a/su.c
+++ b/su.c
@@ -74,8 +74,8 @@ main(int argc, char *argv[])
 	if (setuid(pw->pw_uid) < 0)
 		eprintf("setuid:");
 
-	shell = pw->pw_shell[0] == '\0' ? "/bin/sh" : pw->pw_shell;
 	if (lflag) {
+		shell = pw->pw_shell[0] == '\0' ? "/bin/sh" : pw->pw_shell;
 		term = getenv("TERM");
 		clearenv();
 		setenv("HOME", pw->pw_dir, 1);
@@ -92,6 +92,7 @@ main(int argc, char *argv[])
 		if (pflag) {
 			shell = getenv("SHELL");
 		} else {
+			shell = pw->pw_shell[0] == '\0' ? "/bin/sh" : pw->pw_shell;
 			setenv("HOME", pw->pw_dir, 1);
 			setenv("SHELL", shell, 1);
 			if (strcmp(pw->pw_name, "root") != 0) {
-- 
2.40.1

