--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -532,8 +532,8 @@
         DESTINATION "${SM_INSTALL_DESTINATION}")
 install(DIRECTORY "${SM_ROOT_DIR}/NoteSkins"
         DESTINATION "${SM_INSTALL_DESTINATION}")
-install(FILES "${SM_ROOT_DIR}/Songs/instructions.txt"
-        DESTINATION "${SM_INSTALL_DESTINATION}/Songs")
+install(DIRECTORY "${SM_ROOT_DIR}/Songs"
+        DESTINATION "${SM_INSTALL_DESTINATION}")
 
 if(WITH_CLUB_FANTASTIC)
   # Fetch ClubFantastic packs
