--- a/1
+++ b/1
@@ -1,11 +1,11 @@
 #!/bin/sh
 # vim: set ts=4 sw=4 et:
 
-PATH=/usr/bin:/usr/sbin
+PATH=/usr/bin:/usr/sbin:/bin:/sbin
 
 . /etc/runit/functions
 
-msg "Welcome to Void!"
+msg "Welcome to Gentoo!"
 
 [ -r /etc/rc.conf ] && . /etc/rc.conf
 
--- a/2
+++ b/2
@@ -1,7 +1,7 @@
 #!/bin/sh
 # vim: set ts=4 sw=4 et:
 
-PATH=/usr/bin:/usr/sbin
+PATH=/usr/bin:/usr/sbin:/bin:/sbin
 
 runlevel=default
 for arg in $(cat /proc/cmdline); do
--- a/3
+++ b/3
@@ -1,7 +1,7 @@
 #!/bin/sh
 # vim: set ts=4 sw=4 et:
 
-PATH=/usr/bin:/usr/sbin
+PATH=/usr/bin:/usr/sbin:/bin:/sbin
 
 . /etc/runit/functions
 detect_virt
--- a/core-services/03-filesystems.sh
+++ b/core-services/03-filesystems.sh
@@ -35,7 +35,7 @@
     fi
 fi
 
-if [ -x /usr/bin/zpool -a -x /usr/bin/zfs ]; then
+if [ -x /sbin/zpool -a -x /sbin/zfs ]; then
     if [ -e /etc/zfs/zpool.cache ]; then
         msg "Importing cached ZFS pools..."
         zpool import -N -a -c /etc/zfs/zpool.cache
--- a/ctrlaltdel
+++ b/ctrlaltdel
@@ -1,6 +1,6 @@
 #!/bin/sh
 
-PATH=/usr/bin:/usr/sbin
+PATH=/usr/bin:/usr/sbin:/bin:/sbin
 MSG="System is going down..."
 
 # We check for this file after receiving a SIGCONT to move to stage3
--- a/halt.c
+++ b/halt.c
@@ -109,21 +109,21 @@
     if (do_force)
       reboot(RB_HALT_SYSTEM);
     else
-      execl("/bin/runit-init", "init", "0", (char*)0);
+      execl("/sbin/runit-init", "init", "0", (char*)0);
     err(1, "halt failed");
     break;
   case POWEROFF:
     if (do_force)
       reboot(RB_POWER_OFF);
     else
-      execl("/bin/runit-init", "init", "0", (char*)0);
+      execl("/sbin/runit-init", "init", "0", (char*)0);
     err(1, "poweroff failed");
     break;
   case REBOOT:
     if (do_force)
       reboot(RB_AUTOBOOT);
     else
-      execl("/bin/runit-init", "init", "6", (char*)0);
+      execl("/sbin/runit-init", "init", "6", (char*)0);
     err(1, "reboot failed");
     break;
   case NOOP:
--- a/modules-load
+++ b/modules-load
@@ -1,7 +1,7 @@
 #!/bin/sh
 # modules-load [-n] [-v] - modules-load.d(5) compatible kernel module loader
 
-export PATH=/bin:/sbin
+export PATH=/usr/bin:/usr/sbin:/bin:/sbin
 
 {
 # Parameters passed as modules-load= or rd.modules-load= in kernel command line.
