--- a/Cargo.toml
+++ b/Cargo.toml
@@ -10,6 +10,6 @@
 serde_json = "1.0.82"
 clap = "3.2.8"
 colored = "2.0.0"
-mpvipc = "1.2.0"
+mpvipc = "1.2.1"
 #mpvipc = { path = "/home/jonas/Documents/Projects/mpvipc" }
 
--- a/src/main.rs
+++ b/src/main.rs
@@ -653,7 +653,7 @@
 
                 Some(("wait-for-prop", wait_for_prop_matches)) => {
                     let property = wait_for_prop_matches.value_of("property").unwrap();
-                    mpv.observe_property(&1, property).unwrap();
+                    mpv.observe_property(1, property).unwrap();
                     let mut mpv = mpv;
                     loop {
                         match mpv.event_listen() {
@@ -685,7 +685,7 @@
                     let observed_properties = observe_matches.value_of("properties").unwrap();
                     let props: Vec<&str> = observed_properties.split(',').collect();
                     for (i, property) in props.iter().enumerate() {
-                        mpv.observe_property(i as usize + 1, property).unwrap();
+                        mpv.observe_property(i as isize + 1, property).unwrap();
                     }
                     let mut mpv = mpv;
                     loop {
@@ -756,7 +756,7 @@
                 }
 
                 Some(("raw", _)) => {
-                    mpv.observe_property(99usize, "duration").unwrap();
+                    mpv.observe_property(99isize, "duration").unwrap();
                     let mut mpv = mpv;
                     loop {
                         let event = mpv.event_listen_raw();
