--- a/src/main.rs
+++ b/src/main.rs
@@ -488,9 +488,9 @@
                     }
 
                     if input_str.contains("%position%") {
-                        match mpv.get_property_string("playlist-pos") {
+                        match mpv.get_property::<usize>("playlist-pos") {
                             Ok(position) => {
-                                output_string = output_string.replace("%position%", &position);
+                                output_string = output_string.replace("%position%", &(position + 1).to_string());
                             }
                             Err(msg) => error!("Error: {}", msg),
                         }
@@ -956,13 +956,13 @@
                         let Playlist(entries) = playlist;
                         for entry in entries.iter() {
                             if &entry.title == "" {
-                                let mut output = format!("{}\t{}", entry.id, entry.filename);
+                                let mut output = format!("{}\t{}", entry.id + 1, entry.filename);
                                 if entry.current {
                                     output = format!("{}", output.reverse());
                                 }
                                 println!("{}", output);
                             } else {
-                                let mut output = format!("{}\t{}", entry.id, entry.title);
+                                let mut output = format!("{}\t{}", entry.id + 1, entry.title);
                                 if entry.current {
                                     output = format!("{}", output.reverse());
                                 }
