--- a/src/main.rs
+++ b/src/main.rs
@@ -238,6 +238,7 @@
                 .visible_alias("load")
                 .arg(Arg::with_name("file")
                     .value_name("FILE")
+                    .multiple(true)
                     .required(true))
                 .arg(Arg::with_name("mode")
                     .short('m')
@@ -786,25 +787,26 @@
         Some(("playlist", playlist_matches)) => {
             match playlist_matches.subcommand() {
                 Some(("add", add_matches)) => {
-                    let file = add_matches.value_of("file").unwrap();
-                    let file_type = match add_matches.value_of("type").unwrap() {
-                        "file" => PlaylistAddTypeOptions::File,
-                        "playlist" => PlaylistAddTypeOptions::Playlist,
-                        _ => unreachable!(),
-                    };
-                    let add_option = match add_matches.value_of("mode").unwrap() {
-                        "replace" => PlaylistAddOptions::Replace,
-                        "append" => PlaylistAddOptions::Append,
-                        _ => unreachable!(),
+                    for file in add_matches.values_of("file").unwrap() {
+                        let file_type = match add_matches.value_of("type").unwrap() {
+                            "file" => PlaylistAddTypeOptions::File,
+                            "playlist" => PlaylistAddTypeOptions::Playlist,
+                            _ => unreachable!(),
+                        };
+                        let add_option = match add_matches.value_of("mode").unwrap() {
+                            "replace" => PlaylistAddOptions::Replace,
+                            "append" => PlaylistAddOptions::Append,
+                            _ => unreachable!(),
+                        };
+                        if let Err(msg) = mpv.playlist_add(
+                            file,
+                            file_type,
+                            add_option,
+                        )
+                        {
+                            error!("Error: {}", msg);
+                        }
                     };
-                    if let Err(msg) = mpv.playlist_add(
-                        file,
-                        file_type,
-                        add_option,
-                    )
-                    {
-                        error!("Error: {}", msg);
-                    }
                 }
 
                 Some(("shuffle", _)) => {
