--- a/src/lib.rs
+++ b/src/lib.rs
@@ -90,6 +90,7 @@
 pub enum PlaylistAddOptions {
     Replace,
     Append,
+    AppendPlay,
 }
 
 pub enum PlaylistAddTypeOptions {
@@ -472,8 +473,9 @@
                 &[
                     file.as_ref(),
                     match option {
-                        PlaylistAddOptions::Append => "append",
                         PlaylistAddOptions::Replace => "replace",
+                        PlaylistAddOptions::Append => "append",
+                        PlaylistAddOptions::AppendPlay => "append-play",
                     },
                 ],
             ),
@@ -483,8 +485,9 @@
                 &[
                     file.as_ref(),
                     match option {
-                        PlaylistAddOptions::Append => "append",
                         PlaylistAddOptions::Replace => "replace",
+                        PlaylistAddOptions::Append => "append",
+                        PlaylistAddOptions::AppendPlay => "append-play",
                     },
                 ],
             ),
