--- a/source/frost/messaging/frost/MessageXmlFile.java
+++ b/source/frost/messaging/frost/MessageXmlFile.java
@@ -23,7 +23,7 @@
 import java.util.*;
 import java.util.logging.*;
 
-import org.bouncycastle.util.encoders.*;
+import org.bouncycastle.util.encoders.Base64;
 import org.joda.time.*;
 import org.w3c.dom.*;
 import org.xml.sax.*;
--- a/source/frost/util/FrostCrypt.java
+++ b/source/frost/util/FrostCrypt.java
@@ -37,7 +37,7 @@
 import org.bouncycastle.crypto.params.*;
 import org.bouncycastle.crypto.signers.*;
 import org.bouncycastle.jce.provider.*;
-import org.bouncycastle.util.encoders.*;
+import org.bouncycastle.util.encoders.Base64;
 
 /**
  * Implementation of the crypto layer.
