diff --git a/services/device/usb/usb_context.cc b/services/device/usb/usb_context.cc
index 304be8f14..9e8b82b10 100644
--- a/services/device/usb/usb_context.cc
+++ b/services/device/usb/usb_context.cc
@@ -58,7 +58,11 @@ void UsbContext::UsbEventHandler::Run() {
 
 void UsbContext::UsbEventHandler::Stop() {
   base::subtle::Release_Store(&running_, 0);
+#ifdef LIBUSB_API_VERSION >= 0x01000105
+  libusb_interrupt_event_handler(context_);
+#else
   libusb_interrupt_handle_event(context_);
+#endif
 }
 
 UsbContext::UsbContext(PlatformUsbContext context) : context_(context) {
