--- a/script/rc.conf.in
+++ b/script/rc.conf.in
@@ -27,6 +27,10 @@
 
 # HAVE_CONTROLLER1_GROUPS="true"
 
+# This switch controls the resulting behavior when ctrlaltdel is executed
+# Set REBOOT to "true" to reboot instead of halt
+# REBOOT=""
+
 # The following options control the static network configuration
 #   - NET_INTERFACE: name of device (required)
 #   - NET_ADDRESS: IP address (required)
