--- /dev/null
+++ b/core-services/06-net.sh
@@ -0,0 +1,9 @@
+# vim: set ts=4 sw=4 et:
+
+if [ -n "$NET_INTERFACE" ] && [ -n "$NET_ADDRESS" ] && [ -n "$NET_GATEWAY" ]; then
+    msg "Setting up network interface..."
+    ip link set dev "$NET_INTERFACE" up && \
+    ip addr add "$NET_ADDRESS/${NET_NETMASK:-24}" \
+        brd "${NET_BROADCAST:-+}" dev "$NET_INTERFACE" && \
+    ip route add default via "$NET_GATEWAY"
+fi
--- a/rc.conf
+++ b/rc.conf
@@ -27,3 +27,10 @@
 
 # Amount of ttys which should be setup.
 #TTYS=
+
+# Static networking settings
+#NET_INTERFACE=
+#NET_ADDRESS=
+#NET_NETMASK=
+#NET_BROADCAST=
+#NET_GATEWAY=
