--- a/make.lua
+++ b/make.lua
@@ -1,6 +1,5 @@
 local lm = require 'luamake'
 
-lm.bindir = "bin"
 lm.c = lm.compiler == 'msvc' and 'c89' or 'c11'
 lm.cxx = 'c++17'
 
@@ -15,9 +14,6 @@
     }
 end
 
----@diagnostic disable-next-line: codestyle-check
-lm.EXE_DIR = ""
-
 local includeCodeFormat = true
 
 require "make.detect_platform"
@@ -58,9 +54,17 @@
     }
 }
 
+local platform = require 'bee.platform'
+local exe      = platform.OS == 'Windows' and ".exe" or ""
+
+lm:copy "copy_lua-language-server" {
+    input = lm.bindir .. "/lua-language-server" .. exe,
+    output = "bin/lua-language-server" .. exe,
+}
+
 lm:copy "copy_bootstrap" {
     input = "make/bootstrap.lua",
-    output = lm.bindir .. "/main.lua",
+    output = "bin/main.lua",
 }
 
 lm:msvc_copydll 'copy_vcrt' {
@@ -71,6 +75,7 @@
 lm:phony "all" {
     deps = {
         "lua-language-server",
+        "copy_lua-language-server",
         "copy_bootstrap",
     },
     windows = {
@@ -87,11 +92,8 @@
     return
 end
 
-local platform = require 'bee.platform'
-local exe      = platform.OS == 'Windows' and ".exe" or ""
-
 lm:rule "runtest" {
-    lm.bindir .. "/lua-language-server" .. exe, "$in",
+    "bin/lua-language-server" .. exe, "$in",
     description = "Run test: $in.",
     pool = "console",
 }
