diff -Naur zscroll/setup.py zscroll/setup.py
--- zscroll/setup.py	2015-11-11 17:08:23.000000000 -0500
+++ zscroll/setup.py	2019-03-22 19:12:15.477628224 -0400
@@ -11,7 +11,6 @@
       # to the bin
       scripts=['zscroll'],
       data_files=[('share/man/man1', ['zscroll.1']),
-                  ('share/licenses/zscroll', ['LICENSE']),
                   ('share/zsh/site-functions', ['completion/_zscroll'])
                   ]
       )
