--- a/configure
+++ b/configure
@@ -1856,7 +1856,7 @@
   --without-libintl-prefix     don't search for libintl in includedir and libdir
   --with-manualdir=DIR    Manual directory
   --with-passcrypt-key=KEY     Key used to encode passwords (8 byte string)
-  --with-config-dir=RCDIR      Local configuration dir (default: .claws-mail)
+  --with-config-dir=RCDIR      Local configuration dir (default: .config/claws-mail)
   --with-password-encryption=PROVIDER    Which cryptographic library to use for encrypting stored passwords (gnutls, old, default)
   --with-gpgme-prefix=PFX prefix where GPGME is installed (optional)
 
@@ -22760,7 +22760,7 @@
 	if test x"$platform_win32" = xyes; then
 		ac_cv_with_config_dir="Claws-mail"
 	else
-		ac_cv_with_config_dir=".claws-mail"
+		ac_cv_with_config_dir=".config/claws-mail"
 	fi
 fi
 
--- a/configure.ac
+++ b/configure.ac
@@ -525,7 +525,7 @@
 AC_SUBST(PASSCRYPT_KEY, $with_passcrypt_key)
 
 dnl RC dir (will be default at a certain point in time)
-AC_ARG_WITH(config-dir,    [  --with-config-dir=RCDIR      Local configuration dir (default: .claws-mail)],
+AC_ARG_WITH(config-dir,    [  --with-config-dir=RCDIR      Local configuration dir (default: .config/claws-mail)],
 	      ac_cv_with_config_dir="$withval", ac_cv_with_config_dir="")
 
 dnl Set correct default value based on platform
@@ -533,7 +533,7 @@
 	if test x"$platform_win32" = xyes; then
 		ac_cv_with_config_dir="Claws-mail"
 	else
-		ac_cv_with_config_dir=".claws-mail"
+		ac_cv_with_config_dir=".config/claws-mail"
 	fi
 fi
 AC_DEFINE_UNQUOTED(CFG_RC_DIR, "$ac_cv_with_config_dir", Configuration directory)
