--- a/generator/console.go
+++ b/generator/console.go
@@ -14,7 +14,7 @@
 )
 
 // path to console fonts, adjust it to your distro (e.g. Fedora uses /usr/lib/kbd/consolefonts path for it)
-var consolefontsDir = "/usr/share/kbd/consolefonts/"
+var consolefontsDir = "/usr/share/consolefonts/"
 
 func (img *Image) enableVirtualConsole(vConsolePath, localePath string) (*VirtualConsole, error) {
 	debug("enabling virtual console")
